modules/credit-card-types.ts

```diff
@@ -142,31 +142,50 @@ const cardTypes: CreditCardInformation[] = [
   },
 ];
 
-function matches(value: string, pattern: CreditCardPattern): boolean {
+/**
+ * Checks if a card number matches with a provided pattern.
+ * @param cardNumber - The card number to check for a match.
+ * @param pattern - The pattern used for comparison.
+ * @returns {boolean}
```

Comment:
Tu n'as pas besoin de préciser le type de retour car on utilises Typescript, il est déduit directement
```suggestion
 * @returns
```